### PR TITLE
Get rid of reflection in ObjectSnapshot

### PR DESCRIPTION
**Pull request by Taluu**
Re-targeting of  #34

### CODE REVIEW

File: src/Snapshot/ObjectSnapshot.php [hunk 1]
```diff
@@ -41,18 +41,14 @@ public function __construct($object)
...
-        }
+        $export = (array) $object;
+        $class = get_class($object);
 
-        /** @var \ReflectionProperty $reflProperty */
-        foreach ($refl->getProperties() as $reflProperty) {
-            $reflProperty->setAccessible(true);
-            $value = $reflProperty->getValue($object);
+        foreach ($export as $property => $value) {
+            $property = str_replace(["\x00*\x00", "\x00{$class}\x00"], '', $property); // not accessible properties
```
Comment (lunika) [inline src/Snapshot/ObjectSnapshot.php]: This will be always true ? Can this change in php7 for example ?

Comment (Taluu) [inline src/Snapshot/ObjectSnapshot.php]: Yup. Afterall, the tests are passing with php7 and even hhvm, so...

Comment (lunika) [inline src/Snapshot/ObjectSnapshot.php]: ok so let's go we will see in few year if it's still the case.

**Comment by lunika** [comment]
Looks good

**Comment by Taluu** [comment]
I'll merge that on master when I'll have some time

File: src/Snapshot/ObjectSnapshot.php (hunk 1 above)
Comment (krichprollsch) [inline src/Snapshot/ObjectSnapshot.php]: weird. @Taluu where are you find this hack ?

**Comment by krichprollsch** [comment]
It's also work with inherited attributes?

File: src/Snapshot/ObjectSnapshot.php (hunk 1 above)
Comment (Taluu) [inline src/Snapshot/ObjectSnapshot.php]: during a simple `(array) $object` export :}

Comment (Taluu) [inline src/Snapshot/ObjectSnapshot.php]: and some other things can't remember which too (I think it was in the var_dumper component of symfony)

Comment (Taluu) [inline src/Snapshot/ObjectSnapshot.php]: Maybe I should check up with https://github.com/symfony/var-dumper/blob/master/Caster/Caster.php though, but then I would reintroduce some reflection...

**Pull request merged**